Tests/Pcap++Test/Tests/TcpReassemblyTests.cpp

```diff
@@ -1054,3 +1056,68 @@ PTF_TEST_CASE(TestTcpReassemblyMaxSeq)
 	std::string expectedReassemblyData = readFileIntoString(std::string("PcapExamples/one_tcp_stream_output.txt"));
 	PTF_ASSERT_EQUAL(expectedReassemblyData, stats.begin()->second.reassembledData);
 } //TestTcpReassemblyMaxSeq
+
+PTF_TEST_CASE(TestTcpReassemblyDBOOOBufferCC) // TestTcpReassemblyDisableBaseOutOfOrderBufferClearCondition
+{
+	std::string errMsg;
+	std::vector<pcpp::RawPacket> packetStream;
+	TcpReassemblyMultipleConnStats results1;
+	TcpReassemblyMultipleConnStats results2;
+	pcpp::TcpReassemblyConfiguration config1(true, 5, 30,20,true);
+	pcpp::TcpReassemblyConfiguration config2(true, 5, 30,20,false);
+	pcpp::TcpReassembly tcpReassembly1(tcpReassemblyMsgReadyCallback, &results1, tcpReassemblyConnectionStartCallback, tcpReassemblyConnectionEndCallback, config1);
+	pcpp::TcpReassembly tcpReassembly2(tcpReassemblyMsgReadyCallback, &results2, tcpReassemblyConnectionStartCallback, tcpReassemblyConnectionEndCallback, config2);
+	PTF_ASSERT_TRUE(readPcapIntoPacketVec("PcapExamples/one_tcp_stream.pcap", packetStream, errMsg));
+
+	// unserting a data packet from reverse direction b/w swap 2 consequent data packets
+	std::swap(packetStream[12], packetStream[13]);
+	std::swap(packetStream[13], packetStream[18]);
+
+	TcpReassemblyMultipleConnStats tcpReassemblyResults;
+	for (std::vector<pcpp::RawPacket>::iterator iter = packetStream.begin(); iter != packetStream.end(); iter++)
+	{
+		pcpp::Packet packet(&(*iter));
+		tcpReassembly1.reassemblePacket(packet);
+		tcpReassembly2.reassemblePacket(packet);
+	}
+
+	tcpReassembly1.closeAllConnections();
+	tcpReassembly2.closeAllConnections();
+
+	TcpReassemblyMultipleConnStats::Stats &stats1 = results1.stats;
+	TcpReassemblyMultipleConnStats::Stats &stats2 = results2.stats;
+	PTF_ASSERT_EQUAL(stats1.size(), 1);
+	PTF_ASSERT_EQUAL(stats2.size(), 1);
+	PTF_ASSERT_EQUAL(stats1.begin()->second.numOfDataPackets, 18);
+	PTF_ASSERT_EQUAL(stats2.begin()->second.numOfDataPackets, 19);
+
+	packetStream.clear();
+	tcpReassemblyResults.clear();
+} // TestTcpReassemblyDBOOOBufferCC
+
+PTF_TEST_CASE(TestTcpReassemblyTstamps) // TestTcpReassemblyTimeStamps
+{
+	std::string errMsg;
+	std::vector<pcpp::RawPacket> packetStream;
+
+	PTF_ASSERT_TRUE(readPcapIntoPacketVec("PcapExamples/unidirectional_tcp_stream_with_missing_packet.pcap", packetStream, errMsg));
+
+	TcpReassemblyMultipleConnStats tcpReassemblyResults;
+	tcpReassemblyTest(packetStream, tcpReassemblyResults, true, true);
+
+	TcpReassemblyMultipleConnStats::Stats &stats = tcpReassemblyResults.stats;
+	PTF_ASSERT_EQUAL(stats.begin()->second.numOfDataPackets,7);
+	std::ifstream expectedOutput("PcapExamples/timestamp_output.txt");
+	for(long unsigned int i = 0;i<tcpReassemblyResults.timestamps.size();i++){
+		timeval t = tcpReassemblyResults.timestamps[i];
+		std::string expected;
+		expectedOutput>>expected;
+		int expUsec = std::stoll(expected)%1000000;
+		int expSec = std::stoll(expected)/1000000;
```

Comment:
`std:stoll` is C++11 syntax which PcapPlusPlus hasn't yet officially moved to. We plan to work on it in the coming months. Can we use `atoll` for now and put `std:stoll` in a `TODO C++11` comment?